test/integration/offinstall_int_test.go

```diff
@@ -0,0 +1,224 @@
+package integration
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/ActiveState/archiver"
+	"github.com/ActiveState/cli/internal/constants"
+	"github.com/ActiveState/cli/internal/environment"
+	"github.com/ActiveState/cli/internal/exeutils"
+	"github.com/ActiveState/cli/internal/fileutils"
+	"github.com/ActiveState/cli/internal/subshell/cmd"
+	"github.com/ActiveState/cli/internal/testhelpers/e2e"
+	"github.com/ActiveState/cli/internal/testhelpers/tagsuite"
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/suite"
+)
+
+type OffInstallIntegrationTestSuite struct {
+	tagsuite.Suite
+
+	installerPath   string
+	uninstallerPath string
+}
+
+func (suite *OffInstallIntegrationTestSuite) TestInstallAndUninstall() {
+	suite.OnlyRunForTags(tagsuite.OffInstall)
+
+	// Clean up env after test (windows only for now)
+	if runtime.GOOS == "windows" {
+		env := cmd.NewCmdEnv(true)
+		origPath, err := env.Get("PATH")
+		suite.Require().NoError(err)
+		defer func() {
+			suite.Require().NoError(env.Set("PATH", origPath))
+		}()
+	}
+
+	ts := e2e.New(suite.T(), true)
+	defer ts.Close()
+
+	fmt.Printf("Work dir: %s\n", ts.Dirs.Work)
+
+	suite.preparePayload(ts)
+	targetDir := filepath.Join(ts.Dirs.Work, "target")
+
+	{ // Install
+		tp := ts.SpawnCmdWithOpts(
+			suite.installerPath,
+			e2e.WithArgs(targetDir),
+			e2e.AppendEnv("ACTIVESTATE_CLI_DISABLE_RUNTIME=false"),
+		)
+		tp.Expect("Stage 1 of 3 Finished")
+		tp.Expect("Do you accept the ActiveState License")
+		tp.SendLine("")
+		tp.Expect("Stage 2 of 3 Finished")
+		tp.Expect("Stage 3 of 3 Finished")
+		tp.Expect("Setup environment for installed project?")
+		tp.Send("Y")
+		tp.ExpectExitCode(0)
+
+		// Ensure shell env is updated
+		suite.assertShellUpdated(targetDir, true)
+
+		// Ensure installation dir looks correct
+		suite.assertInstallDir(targetDir, true)
+
+		// Run executable and validate that it has the relocated value
+		if runtime.GOOS == "windows" {
+			refreshEnv := filepath.Join(environment.GetRootPathUnsafe(), "test", "integration", "testdata", "tools", "refreshenv", "refreshenv.bat")
+			tp = ts.SpawnCmd("cmd", "/C", refreshEnv+" && test-offline-install")
+		} else {
+			tp = ts.SpawnCmd("bash", "-c", "test-offline-install")
+		}
+		tp.Expect("TEST REPLACEMENT", 5*time.Second)
+		tp.ExpectExitCode(0)
+	}
+
+	{ // Uninstall
+		tp := ts.SpawnCmdWithOpts(
+			suite.uninstallerPath,
+			e2e.WithArgs(targetDir),
+			e2e.AppendEnv("ACTIVESTATE_CLI_DISABLE_RUNTIME=false"),
```

Comment:
Does the uninstaller need a runtime enabled?

Reply:
Yes, it needs to access the runtime in order to find out what files to delete and whatnot.